src/pudl/analysis/pudl_models.py

```diff
@@ -39,13 +43,121 @@ def core_sec10k__quarterly_company_information() -> pd.DataFrame:
+        values="company_information_fact_value",
+        index=["filename_sec10k", "report_date"],
+        columns="company_information_fact_name",
+        aggfunc="first",
+    )
+    df.columns.name = None
+    df = df.reset_index()
+    # we want central_index_key and report_date to be a primary key
+    # prioritize records where the filer is the same
+    # as the harvested central index key value
+    df["filer_cik"] = df["filename_sec10k"].str.split("/").str[2].str.zfill(10)
+    df["filer_cik_matches_cik"] = df["filer_cik"] == df["central_index_key"]
+    df = df.sort_values(by="filer_cik_matches_cik", ascending=False).drop_duplicates(
+        subset=["central_index_key", "report_date"], keep="first"
+    )
+    df = df.drop(columns=["filer_cik", "filer_cik_matches_cik"])
+    df = df.rename(
+        columns={
+            "street_1": "street_address",
+            "street_2": "address_2",
+            "company_conformed_name": "company_name",
+            "date_of_name_change": "name_change_date",
+            "zip": "zip_code",
+            "business_phone": "phone_number",
+            "irs_number": "company_id_irs",
+            "former_conformed_name": "company_name_former",
+            "form_type": "sec10k_version",
+            "standard_industrial_classification": "industry_id_sic",
+        }
+    )
+
+    return df
+
+
+@asset(
+    io_manager_key="pudl_io_manager",
+    group_name="pudl_models",
+    ins={
+        "core_df": AssetIn("core_sec10k__quarterly_company_information"),
+        "eia_utils_df": AssetIn("core_eia__entity_utilities"),
+    },
+)
+def out_sec10k__quarterly_company_information(
+    core_df: pd.DataFrame, eia_utils_df: pd.DataFrame
+) -> pd.DataFrame:
+    """Company information extracted from SEC10k filings and matched to EIA utilities."""
+    matched_df = _load_table_from_gcs("out_sec10k__parents_and_subsidiaries")
+    matched_df = (
+        matched_df[["central_index_key", "utility_id_eia"]]
+        .dropna()
+        .drop_duplicates(
+            subset="central_index_key"
+        )  # matches should already be 1-to-1 but drop duplicates to ensure this is true
+    )
+    out_df = core_df.merge(matched_df, how="left", on="central_index_key")
+    # merge utility name on
+    out_df = out_df.merge(eia_utils_df, how="left", on="utility_id_eia")
+    return out_df
+
+
+@asset(
+    io_manager_key="pudl_io_manager",
+    group_name="pudl_models",
+    ins={
+        "core_df": AssetIn("core_sec10k__quarterly_company_information"),
+    },
+)
+def core_sec10k__changelog_company_name(
+    core_df: pd.DataFrame,
+) -> pd.DataFrame:
+    """Changes in SEC company names and the date of change as reported in 10k filings."""
+    changelog_df = core_df[
+        [
+            "central_index_key",
+            "report_date",
+            "company_name",
+            "name_change_date",
+            "company_name_former",
+        ]
+    ]
```

Comment:
Maybe I misunderstood this table, but basically it's just pulling the recorded name changes and dates out of the core (pivoted) table? Or I guess this could be made more robust by actually tracking when the name changes for each CIK even if the company doesn't report a name change.

Reply:
I was imagining this just representing the name changes as they're explicitly reported -- not checking against all the report dates and company names that are observed to try and detect changes.  That's a good idea! But might not be worth the effort, and in any case we would first need a straightforward extraction of the reported name changes to check against the empirical name changes.